umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj

```diff
@@ -81,17 +92,32 @@
     (into {} (for [[k ^String v] (partition 2 (str/split orbit-string #":? "))]
                [(keyword k) (Double/parseDouble v)]))))
 
+(defn- parse-vertical-domains
+  "Parse the vertical domain from the ISO XML document. Vertical domains are encoded in an ISO XML
+  document as a single string like this:
+  \"Type: Some type Value: Some value\""
+  [doc]
+  (when-let [vertical-domains (select doc vertical-string-xpath)]
+    (for [vertical-domain vertical-domains]
```

Comment:
That code is all totally fine. In the code you pasted that didn't work, in your `:let` you are using vertical-domains again. Don't change it in the when-let, change it in the :let and :when.